### PR TITLE
feat(chat): enable atomic tools

### PR DESCRIPTION
**Pull request by rachmlenig**
## Summary by Sourcery

Enable atomic tool support in the chat agent by refactoring the inference router to delegate processing to a new services layer, introducing a dynamic tool registry and ProjectsTool for project management, and modularizing message analysis and response validation strategies. Add comprehensive tests and documentation to support the new tool architecture.

New Features:
- Integrate atomic tool architecture into chat inference, enabling native and manual tool execution
- Add ProjectsTool as an atomic tool for listing and creating projects via chat
- Implement modular analysis and validation strategies using rule-based and LLM approaches

Enhancements:
- Refactor the inference router to use ChatProcessor and AgentSessionManager for cleaner session handling
- Introduce ToolRegistry and base tool abstractions under tools/core for dynamic tool discovery
- Modularize chat processing into services (analyzers, factories, response formatting, and tool execution)

Documentation:
- Add README documentation for the ProjectsTool usage and integration

Tests:
- Add comprehensive tests for the chat endpoint covering positive, negative, and edge cases
- Add unit tests for the ProjectsTool functionality

### CODE REVIEW

**Comment by rachmlenig** [comment]
@sourcery-ai review

**Comment by rachmlenig** [comment]
@sourcery-ai dismiss

**Comment by rachmlenig** [comment]
@sourcery-ai guide

**Comment by rachmlenig** [comment]
@sourcery-ai review

File: server/api/routers/inference/analyzers.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        """
+        Use LLM to analyze user intent for project-related actions.
+        Falls back to rule-based analysis if LLM is unavailable.
+        """
+        if not self.client:
+            return self._fallback_analysis(message)
+        
+        try:
+            system_prompt = """
+You are an expert at analyzing user messages to determine project management actions.
```
Comment (mhamann) [inline server/api/routers/inference/analyzers.py:89]: The LLM might infer this to mean "project manager" type of things (like in a work stream). I wonder if we need to say something like "You are an expert in building AI pipelines that include prompt engineering, rag, and model tuning."

Comment (BobbyRadford) [reply server/api/routers/inference/analyzers.py:89]: Try using AI to build a system prompt

**Comment by mhamann** [comment]
Going to merge this one since it's essentially additive. We can iterate to improve the system prompts.
